src/viam/sdk/components/camera/camera.cpp

```diff
@@ -24,6 +25,96 @@ API API::traits<Camera>::api() {
     return {kRDK, kComponent, "camera"};
 }
 
+// Appends a uint64_t value in big-endian format to a byte vector and updates the offset.
+void append_uint64_big_endian(std::vector<unsigned char>& data, size_t& offset, uint64_t value) {
+    const uint64_t value_be = boost::endian::native_to_big(value);
+    if (data.size() < offset + 8) {
+        throw Exception("Incorrect data size-- attempted to write beyond data bounds");
+    }
+    std::memcpy(&data[offset], &value_be, 8);
+    offset += 8;
+}
+
+// Reads a uint64_t value from data in big-endian format and updates the offset.
+// Intended to be used in a sequential manner.
+uint64_t read_uint64_big_endian(const std::vector<unsigned char>& data, size_t& offset) {
+    if (data.size() < offset + 8) {
+        throw Exception("Attempted to read beyond data bounds.");
+    }
+
+    uint64_t value;
+    std::memcpy(&value, &data[offset], 8);
+    value = boost::endian::big_to_native(value);
+    offset += 8;
+
+    return value;
+}
+
+std::vector<unsigned char> Camera::encode_depth_map(const Camera::depth_map& m) {
+    if (m.depth_values.size() != m.width * m.height) {
+        throw Exception(
+            "Number of depth values does not match the specified width and height. Expected: " +
+            std::to_string(m.width * m.height) +
+            ". Actual: " + std::to_string(m.depth_values.size()));
+    }
+
+    const size_t total_byte_count = HEADER_SIZE + m.depth_values.size() * sizeof(uint16_t);
+    std::vector<unsigned char> data(total_byte_count);
+    size_t offset = 0;
+
+    // Network data is stored in big-endian, while most host systems are little endian.
+    append_uint64_big_endian(data, offset, MAGIC_NUMBER);
```

Comment:
Might be better if `offset` was passed by pointer here to make it clearer to a reader that it was going to be modified.